src/main/java/de/metas/ui/web/handlingunits/process/WEBUI_M_HU_MoveToAnotherWarehouse_Helper.java

```diff
@@ -0,0 +1,124 @@
+package de.metas.ui.web.handlingunits.process;
+
+import java.util.Comparator;
+import java.util.List;
+
+import org.adempiere.util.Services;
+
+import com.google.common.collect.ImmutableList;
+
+import de.metas.handlingunits.IHUStatusBL;
+import de.metas.handlingunits.IHandlingUnitsDAO;
+import de.metas.handlingunits.attribute.IHUAttributesDAO;
+import de.metas.handlingunits.model.I_M_HU;
+import de.metas.handlingunits.model.I_M_Warehouse;
+import de.metas.handlingunits.movement.api.HUMovementResult;
+import de.metas.handlingunits.movement.api.IHUMovementBL;
+import de.metas.process.IProcessPrecondition;
+import de.metas.process.Param;
+import de.metas.process.ProcessPreconditionsResolution;
+import de.metas.ui.web.handlingunits.HUEditorProcessTemplate;
+import de.metas.ui.web.handlingunits.HUEditorRowFilter.Select;
+import de.metas.ui.web.handlingunits.WEBUI_HU_Constants;
+import de.metas.ui.web.process.descriptor.ProcessParamLookupValuesProvider;
+import de.metas.ui.web.window.datatypes.LookupValue.IntegerLookupValue;
+import de.metas.ui.web.window.datatypes.LookupValuesList;
+import de.metas.ui.web.window.descriptor.DocumentLayoutElementFieldDescriptor.LookupSource;
+
+/*
+ * #%L
+ * metasfresh-webui-api
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+/**
+ * This is the common structure for a process that moves HUs from a warehouse to another.
+ * Is extended by specific processes, depending of their particular requirements
+ *
+ * @author metas-dev <dev@metasfresh.com>
+ *
+ */
+public class WEBUI_M_HU_MoveToAnotherWarehouse_Helper extends HUEditorProcessTemplate implements IProcessPrecondition
+{
+	private final transient IHandlingUnitsDAO handlingUnitsDAO = Services.get(IHandlingUnitsDAO.class);
+	private final transient IHUMovementBL huMovementBL = Services.get(IHUMovementBL.class);
+	private final transient IHUStatusBL huStatusBL = Services.get(IHUStatusBL.class);
+	final IHUAttributesDAO huAttributesDAO = Services.get(IHUAttributesDAO.class);
+
+	private HUMovementResult movementResult = null;
+
+	@Param(parameterName = I_M_Warehouse.COLUMNNAME_M_Warehouse_ID, mandatory = true)
+	private I_M_Warehouse warehouse;
+
+	@Override
+	protected ProcessPreconditionsResolution checkPreconditionsApplicable()
```

Comment:
i suggest to annotate this method with `@OverridingMethodsMustInvokeSuper` to make your intent clearer